GitUI/ScriptEngine/ScriptsManager.cs

```diff
@@ -63,7 +67,20 @@ public static bool RunEventScripts(GitModuleForm form, ScriptEvent scriptEvent)
             return true;
         }
 
-        public static string? SerializeIntoXml()
+        public CommandStatus RunScript<THostForm>(int scriptId, THostForm form, RevisionGridControl? revisionGrid = null)
+            where THostForm : IGitModuleForm, IWin32Window
+        {
+            ScriptInfo? scriptInfo = GetScript(scriptId);
+            if (scriptInfo is null)
+            {
+                throw new UserExternalOperationException($"{TranslatedStrings.ScriptErrorCantFind}: '{scriptId}'",
```

Comment:
Good point. But internally we operate on IDs and not labels (which can be duplicated in the distributed scenario).
As you said, we should never be in this situation; and in unlikely event we end up here is because of the direct user actions.